Lib/defcon/tools/notifications.py

```diff
@@ -401,12 +401,59 @@ def __setitem__(self, key, value):
         super(ObserverDict, self).__setitem__(key, value)
         self._order.append(key)
 
+
+class _ObserverDict2(_ObserverDict):
+
+    """An object for storing ordered observers."""
+
+    def __init__(self):
+        super(_ObserverDict, self).__init__()
+        self._order = []
+
+    def keys(self):
+        return list(self._order)
+
+    def values(self):
+        return [self[key] for key in self]
+
+    def items(self):
+        return [(key, self[key]) for key in self]
+
+    def __iter__(self):
+        order = self._order
+        while order:
+            yield order[0]
+            order = order[1:]
```

Comment:
> ObserverDict is now a subclass of OrderedDict.

This was the case originally and it was changed in 88a113ee4bc14309c509ff317dcedcc993693c3a _because_ stdlib was alledgedly slower than a reimpl., hence why I think this warrants benchmarks.